pkg/api/dashboard.go

```diff
@@ -273,6 +273,69 @@ func (hs *HTTPServer) getDashboardHelper(ctx context.Context, orgID int64, id in
 	return queryResult, nil
 }
 
+func (hs *HTTPServer) RestoreDashboard(c *contextmodel.ReqContext) response.Response {
+	uid := web.Params(c.Req)[":uid"]
+
+	dash, err := hs.DashboardService.GetSoftDeletedDashboard(c.Req.Context(), c.SignedInUser.GetOrgID(), uid)
+	if err != nil {
+		return response.Error(http.StatusNotFound, "Dashboard not found", err)
+	}
+
+	guardian, err := guardian.NewByDashboard(c.Req.Context(), dash, c.SignedInUser.GetOrgID(), c.SignedInUser)
+	if err != nil {
+		return response.Err(err)
+	}
+
+	if canRestore, err := guardian.CanDelete(); err != nil || !canRestore {
```

Comment:
I think this is a good IAM question, but I would presume that you should still check for delete permission when soft deleting.

I guess ideally soft delete and hard delete would be two seperate permissions?

Reply:
That's correct, Soft deleting is checking delete permissions. 
I don't think we need delete permissions here since the endpoint is checking for write permissions (because in the end, it is changing the state of a dashboard).

Reply:
I've found that Restoring a dashboard version requires Dashboard Write permissions and it checks `guardian.CanSave()`, since restoring a deleted dashboard is similar from a user case perspective I think we should check the same permissions.